src/results.jl

```diff
@@ -271,40 +277,43 @@ function execute(
     jl_conn::Connection,
     query::AbstractString;
     throw_error::Bool=true,
+    binary_format::Bool=false,
     kwargs...
 )
     result = lock(jl_conn) do
-        _execute(jl_conn.conn, query)
+        _execute(jl_conn.conn, query; binary_format)
     end
 
-    return handle_result(Result(result, jl_conn; kwargs...); throw_error=throw_error)
+    return handle_result(Result(result, jl_conn; binary_format, kwargs...); throw_error=throw_error)
 end
 
 function execute(
     jl_conn::Connection,
     query::AbstractString,
     parameters::Union{AbstractVector, Tuple};
     throw_error::Bool=true,
+    binary_format::Bool=false,
     kwargs...
 )
     string_params = string_parameters(parameters)
     pointer_params = parameter_pointers(string_params)
 
     result = lock(jl_conn) do
-        _execute(jl_conn.conn, query, pointer_params)
+        _execute(jl_conn.conn, query, pointer_params; binary_format)
     end
 
-    return handle_result(Result(result, jl_conn; kwargs...); throw_error=throw_error)
+    return handle_result(Result(result, jl_conn; binary_format, kwargs...); throw_error=throw_error)
 end
 
-function _execute(conn_ptr::Ptr{libpq_c.PGconn}, query::AbstractString)
-    return libpq_c.PQexec(conn_ptr, query)
-end
+# function _execute(conn_ptr::Ptr{libpq_c.PGconn}, query::AbstractString)
```

Comment:
Don't merge these methods; one calls `PQexec` and one calls `PQexecParams` and that distinction is important as the two functions have different capabilities. There needs to be some API for the user to use the `PQexec` version. Maybe we'll need a separate API function since binary format is not supported with `PQexec`.

Reply:
What if we just required people wanting to use binary format to always pass parameters (even if they're empty)? Or had `execute_binary` instead of having a kwarg? Just brainstorming.

Reply:
With `execute` and `execute_params` I was trying to keep it similar to `libpq_c.PQexec` and `libpq_c.PQexecParams`. I think that makes the most sense but I'm happy to change it to something else.